examples/complicated_cpu_bound_benchmark.cf

```diff
@@ -0,0 +1,1031 @@
+      "b" int => hash_to_int(0, "$(i)", "world");
+      "c" int => hash_to_int(0, "$(i)", "$(sys.key_digest)");
+      "d" int => hash_to_int(0, "$(i)", "$(sys.policy_hub)");
+}
+
+bundle agent ifelse_bench(i)
+{
+  vars:
+      "outcome"
+        string => ifelse(regcmp("[0-9]*[02468]", "$(i)"), "is odd",
+                         regcmp("[0-9]", "$(i)"), "is less than 10",
+                         "is even");
+}
+
+bundle agent intersection_bench(i)
+{
+  vars:
+      "a"
+        slist => { "a", "b", "c", "d", "e", "f" };
+      "b"
+        slist => { "a", "c", "e", "g", "i", "k" };
+      "inter"
+        slist => intersection("a", "b");
+}
+
+bundle agent isgreaterthan_bench(i)
+{
+  classes:
+      "indeed"
+        expression => isgreaterthan("5", "$(i)");
+}
+
+bundle agent islessthan_bench(i)
+{
+  classes:
+      "indeed"
+        expression => islessthan("5", "$(i)");
+}
+
+bundle agent isvariable_bench(i)
+{
+  vars:
+      "defined"
+        string => "this var is defined";
+  classes:
+      "test_1"
+        expression => isvariable("defined");
+      "test_2"
+        expression => isvariable("undefined");
+}
+
+bundle agent join_bench(i)
+{
+  vars:
+      "list"
+        slist => { "a", "b", "c", "d", "e", "e" };
+      "joined_list"
+        string => join(", ", "list");
+      "data"
+        data => '[ "a", "b", "c", "d", "e", "f" ]';
+      "joined_data"
+        string => join("->", "data");
+}
+
+bundle agent laterthan_bench(i)
+{
+  classes:
+      "test"
+        expression => laterthan("$(i)", "$(i)", "$(i)", "$(i)", "$(i)", "$(i)");
+}
+
+bundle agent length_bench(i)
+{
+  vars:
+      "list"
+        slist => { expandrange("[0-$(i)]", "1") };
+      "len"
+        int => length("list");
+}
+
+bundle agent maparray_bench(i)
+{
+  vars:
+      "arr[0]"
+        string => "a";
+      "arr[1]"
+        string => "b";
+      "arr[2]"
+        string => "c";
+      "arr[3]"
+        string => "d";
+      "arr[4]"
+        string => "e";
+      "arr[5]"
+        string => "f";
+      "mapped"
+        slist => maparray("key=$(this.k), val=$(this.v)", "arr");
+}
+
+bundle agent mapdata_bench(i)
+{
+  vars:
+      "data"
+        data => '{ "name": "alice", "age": $(i) }';
+      "map_json"
+        data => mapdata("json", '{ "key": "$(this.k)", "value": "$(this.v)" }', "data");
+}
+
+bundle agent maplist_bench(i)
+{
+  vars:
+      "list"
+        slist => { "a", "b", "c", "d", "e", "f" };
+      "mapped"
+        slist => maplist("Element: $(this)", "list");
+}
+
+bundle agent max_bench(i)
+{
+  vars:
+      "alpha"
+        slist => { "a", "c", "b", "d", "f", "e" };
+      "numer"
+        ilist => { 1, 3, 2, 4, 6, 5 };
+      "alpha_max"
+        string => max("alpha", "lex");
+      "numer_max"
+        string => max("numer", "int");
+}
+
+bundle agent mean_bench(i)
+{
+  vars:
+      "list"
+        slist => { "a", "1", "b", "2", "f", "6" };
+      "mean" 
+        real => mean("list");
```

Comment:
In CFEngine, the string quotes do not matter, but I like to not quote something like this, to indicate that it's a variable name:

```suggestion
        real => mean(list);
```